src/AppInstallerCLICore/ContextOrchestrator.cpp

```diff
@@ -128,22 +126,34 @@ namespace AppInstaller::CLI::Execution
             const auto& manifest = item->GetContext()->Get<Execution::Data::Manifest>();
             m_installingWriteableSource->RemovePackageVersion(manifest, std::filesystem::path{ manifest.Id + '.' + manifest.Version });
 
-            ::SetEvent(item->GetCompletedEvent().get());
+            item->GetCompletedEvent().SetEvent();
         }
     }
-    void ContextOrchestrator::CancelItem(const std::shared_ptr<OrchestratorQueueItem>& item)
+    void ContextOrchestrator::CancelQueueItem(const std::shared_ptr<OrchestratorQueueItem>& item)
     {
         // Always cancel the item, even if it isn't running yet, to get the terminationHR set correctly.
         item->GetContext()->Cancel(false, true);
 
         RemoveItemInState(item, OrchestratorQueueItemState::Queued);
     }
 
-    std::shared_ptr<OrchestratorQueueItem> ContextOrchestrator::GetQueueItem(std::wstring id)
+    std::shared_ptr<OrchestratorQueueItem> ContextOrchestrator::GetQueueItem(const OrchestratorQueueItemId* queueItemId)
     {
         std::lock_guard<std::mutex> lock{ m_queueLock };
 
-        return FindById(id);
+        return FindById(queueItemId);
+    }
+
+    bool OrchestratorQueueItemId::IsSame(const OrchestratorQueueItemId* comparedId)
+    {
+        return ((GetPackageId() == comparedId->GetPackageId()) && 
+                (GetSourceId() == comparedId->GetSourceId()));
+    }
+
+    std::shared_ptr<OrchestratorQueueItem> OrchestratorQueueItemFactory::CreateItemForInstall(std::wstring packageId, std::wstring sourceId, std::shared_ptr<COMContext> context)
+    {
+        std::shared_ptr<OrchestratorQueueItemId> identifier = std::make_shared<Execution::OrchestratorQueueItemId>(std::move(packageId), std::move(sourceId));

```

Comment:
I don't think that these need to be `shared_ptr`, nor do any of the functions need to be taking pointers.  Take const& instead to indicate that `nullptr` is not a valid value.